pytests/threads/multiThreadingLockSimple.py

```diff
@@ -0,0 +1,23 @@
+# Threads acquire a lock and modify global variable
+import thread
+num_threads = 1
+lock = thread.allocate_lock()
+def func(a):
+    global num_threads
+    lock.acquire()
+    print "Num Thread: ", num_threads
+    num_threads += 1
+    lock.release()
+
+thread.start_new_thread(func,(1,))
+thread.start_new_thread(func,(2,))
+thread.start_new_thread(func,(3,))
+thread.start_new_thread(func,(4,))
+thread.start_new_thread(func,(5,))
+thread.start_new_thread(func,(6,))
+
+def incr():
```

Comment:
When any thread encounters a long loop, there is no way to return back to the thread scheduler. It considers the below loop to be a single bytecode method, (which it is), but once it starts executing the loop, the thread scheduler has no way of regaining control (and scheduling other threads) until the loop terminates. 

Because of this, if there are two threads running, and one of them starts running a loop, it will prevent the other thread from being scheduled until it is done with its loop, no matter how long it executes for.

e.g.:
```python
count = 0
while(1):
    count = count + 1
    if (count == 100000):
        break
```

But if you call a function inside of a loop, the counter inside of thread.run is decremented, since another bytecode method is being called, and other threads can be scheduled. @jvilk said that this was a current limitation of Ninia since most programs with long-running loops call at least one function in the loop body which decrements the counter.
https://github.com/plasma-umass/Ninia/blob/master/src/threading.ts#L54